src/types.ts

```diff
@@ -45,7 +53,7 @@ export interface PoolInstance {
     depositId: string,
     lockUntil: ethers.BigNumber,
     signer: ethers.Signer
-  ) => Promise<ethers.ContractTransaction>;
+  ) => Promise<ethers.ContractTransaction | null>;
```

Comment:
does null signify the transaction failing? I figured an error might be thrown.

Reply:
it was `| null` while I was testing in several places. Must have not removed one